--- a/site-packages/scapy/contrib/scada/iec104/iec104_fields.py	2021-01-24 13:50:26.000000000 +0100
+++ b/site-packages/scapy/contrib/scada/iec104/iec104_fields.py	2021-07-27 14:14:40.742417874 +0200
@@ -97,11 +97,11 @@
     """
 
     def __init__(self, name, default):
-        Field.__init__(self, name, default, "!I")
+        Field.__init__(self, name, default, "<H")
 
     def addfield(self, pkt, s, val):
         b0 = (val << 1) & 0xfe
-        b1 = val >> 7
+        b1 = (val >> 7) & 0xff
 
         return s + bytes(bytearray([b0, b1]))
 
